### PR TITLE
osu! client login refactoring, fixes, cleanup

### PR DESCRIPTION
**Pull request by cmyui**
# Describe your changes

## Checklist
- [x] I've manually tested my code

### CODE REVIEW

**Comment by cmyui** [comment]
Oopsie sorry - not yet ready for review

File: app/api/domains/cho.py [hunk 1]
```diff
@@ -636,17 +684,12 @@ async def login(
...
-
+    user_info = await authenticate(login_data["username"], login_data["password_md5"])
     if user_info is None:
-        # no account by this name exists.
         return {
-            "osu_token": "unknown-username",
+            "osu_token": "incorrect-credentials",
             "response_body": (
-                app.packets.notification(f"{BASE_DOMAIN}: Unknown username")
+                app.packets.notification(f"{BASE_DOMAIN}: Incorrect credentials")
```
Comment (cmyui) [inline app/api/domains/cho.py:692]: incorrect username vs. incorrect password will now both return "incorrect credentials"

**Comment by cmyui** [comment]
if anyone has ideas of where to place these new functions -- feel free to suggest. thinking about it myself as well.

File: app/api/domains/cho.py [hunk 2]
```diff
@@ -636,22 +688,17 @@ async def login(
...
                 + app.packets.user_id(-1)
             ),
         }
 
-    if osu_version.stream == "tourney" and not (
+    if osu_version.stream == OsuStream.TOURNEY and not (
```
Comment (cmyui) [inline app/api/domains/cho.py:701]:
```suggestion
    if osu_version.stream is OsuStream.TOURNEY and not (
```

File: app/api/domains/cho.py (hunk 1 above)
Comment (NiceAesth) [reply app/api/domains/cho.py:706]: heh, i halfway agree but this is definitely kinda funny since all profiles are publicly searchable by username

**Pull request merged**